transport/tchannel.go

```diff
@@ -182,6 +200,18 @@ func (t *tchan) writeArgs(call *tchannel.OutboundCall, r *Request) error {
 	return nil
 }
 
+func applyTChanOptions(callOpts *tchannel.CallOptions, opts map[string]string) {
+	if format, ok := opts["as"]; ok {
+		callOpts.Format = tchannel.Format(format)
+	}
+	if rd, ok := opts["rd"]; ok {
+		callOpts.RoutingDelegate = rd
+	}
+	if sk, ok := opts["sk"]; ok {
+		callOpts.ShardKey = sk
+	}
```

Comment:
I would like to, but I'm not sure if it's worth the code. I want to:
 - not mutate the options (since you can create multiple transports with the same options)
 - not have to repeat myself for what options we've checked

I'm also wondering whether unknown options should be propagated as custom transport headers, not sure yet.